# Python fundamentals Practice

#### Create a variable `x` with the value of 6 to the power of 5; print the result

In [2]:
x = 6 ** 5
x

7776

#### Print the following string using the `.format()` method or f-strings, piping in the value of `x`

In [3]:
'6 to the power of 5 is {}'.format(x)

'6 to the power of 5 is 7776'

#### Save this to a variable `words`; how long is this string?

In [4]:
words = '6 to the power of 5 is {}'.format(x)
len(words)

27

#### How many spaces are in this string? Use either `.count()` or `.split()`

In [5]:
words.count(" ")

7

In [6]:
len(words.split(" "))-1

7

#### Slice the string to return just the "7776" portion; save it to a variable "y" and print the result as an integer

In [8]:
y = int(words[23:])
y

7776

#### Replace the first "6" and "5" with "six" and "five" in this string

In [9]:
words = words.replace("6", "six", 1)
words

'six to the power of 5 is 7776'

In [10]:
words = words.replace("5","five",1)
words

'six to the power of five is 7776'

#### Create a list `x` with the values 4, 7, 8, 10, 7, 4, 100; print the result

In [11]:
x = [4, 7, 8, 10, 7, 4, 100]
x

[4, 7, 8, 10, 7, 4, 100]

#### How long is this list?  How many unique values are in this list (hint: use `set()`)

In [12]:
len(x)

7

In [13]:
len(set(x))

5

#### Sort this list and reassign it back to `x`; print the result

In [14]:
x= sorted(x)
x

[4, 4, 7, 7, 8, 10, 100]

#### Append the value `1` to the _end_ of the list; print the result

In [15]:
x.append(1)
x

[4, 4, 7, 7, 8, 10, 100, 1]

#### Report the following slice of this list

In [16]:
x[2:5]

[7, 7, 8]

#### Print the following value of this list

In [17]:
print(x[-1])

1


#### Given the following list, return the `5000` value

In [19]:
nested_list = [0, [1, 2, 3, [4, 5, [6]]], [7, [8, [9, 10], 5000]], 11, [12]]

In [20]:
nested_list[2][1][2]

5000

#### Given the following _dictionary_ return the word "target"

In [21]:
nested_dict = {"a": {"A": "aa"}, "b": ["B"], "c": {"words": ["bullseye", "target"]}}

In [22]:
nested_dict["c"]["words"][1]

'target'

#### Print the keys of this dictionary

In [23]:
nested_dict.keys()

dict_keys(['a', 'b', 'c'])

#### Add a new entry to this dictionary (use any key or value you'd like); print the resulting dictionary

In [24]:
nested_dict["d"] = "new"
nested_dict

{'a': {'A': 'aa'},
 'b': ['B'],
 'c': {'words': ['bullseye', 'target']},
 'd': 'new'}

#### Get and print *just* the value you've stored

In [25]:
nested_dict["d"]

'new'

#### Loop over the list `x` and print out the square root of each value

In [26]:
for value in x:
    print(value ** 0.5)

2.0
2.0
2.6457513110645907
2.6457513110645907
2.8284271247461903
3.1622776601683795
10.0
1.0


#### Perform the same operation, but instead of printing the values, return them as a list (hint: use either an empty list and `.append()` or a list comprehension)

In [27]:
[value ** 0.5 for value in x]

[2.0,
 2.0,
 2.6457513110645907,
 2.6457513110645907,
 2.8284271247461903,
 3.1622776601683795,
 10.0,
 1.0]

#### Create a function that grabs the email website domain from a string in the form: `"jared@jhu.edu"`
    
(So for example, passing `"jared@jhu.edu"` would return: `"jhu.edu"`)

In [36]:
def extract(email):
    print(email.split("@")[-1])

In [38]:
extract("jared@jhu.edu")

jhu.edu


#### Create a function that takes as input a number, and prints out a statement denoting if that number is even or odd

(So, given `2` it might print `even!`)

In [39]:
def check(x):
    if x % 2 ==0:
        print(f"{x} is even!")
    else:
        print(f"{x} is odd!")

In [40]:
check(2)

2 is even!


In [41]:
check(5)

5 is odd!
